flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -297,9 +328,344 @@ private static Object handleLogicalTypeSchema(Schema fieldSchema, Object value)
          */
-        private static UnaryOperator<Object> getLogicalEncoder(String logicalTypeString) {
+        private static UnaryOperator<Object> getLogicalEncoder(
+                Schema fieldSchema, String logicalTypeString) {
             Map<String, UnaryOperator<Object>> mapping = new HashMap<>();
+            mapping.put(LogicalTypes.date().getName(), AvroSchemaHandler::convertDate);
+            mapping.put(
+                    LogicalTypes.decimal(1).getName(),
+                    value -> AvroSchemaHandler.convertBigDecimal(value, fieldSchema));
+            mapping.put(
+                    LogicalTypes.timestampMicros().getName(),
+                    value -> convertTimestamp(value, true, "Timestamp(micros/millis)"));
+            mapping.put(
+                    LogicalTypes.timestampMillis().getName(),
+                    value -> convertTimestamp(value, false, "Timestamp(micros/millis)"));
+            mapping.put(LogicalTypes.uuid().getName(), AvroSchemaHandler::convertUUID);
+            mapping.put(LogicalTypes.timeMillis().getName(), value -> convertTime(value, false));
+            mapping.put(LogicalTypes.timeMicros().getName(), value -> convertTime(value, true));
+            mapping.put(
+                    LogicalTypes.localTimestampMillis().getName(),
+                    value -> convertDateTime(value, false));
+            mapping.put(
+                    LogicalTypes.localTimestampMicros().getName(),
+                    value -> convertDateTime(value, true));
+            mapping.put("geography_wkt", AvroSchemaHandler::convertGeography);
+            mapping.put("Json", AvroSchemaHandler::convertJson);
             return mapping.get(logicalTypeString);
         }
+
+        // ---- Utilities to enable Conversions for Logical Types ------------
+        @VisibleForTesting
+        static String convertUUID(Object value) {
+            if (value instanceof UUID) {
+                return ((UUID) value).toString();
+            } else if (value instanceof String) {
+                UUID.fromString((String) value);
+                return (String) value;
+            }
+            LOG.error(getLogErrorMessage("String/UUID", "UUID", value.getClass().toString()));
+            throw new IllegalArgumentException(getErrorMessage("String/UUID", "UUID"));
+        }
+
+        private static void validateTimestamp(long timestamp) {
+            try {
+                // Validate timestamp from microseconds since EPOCH.
+                Timestamp.ofTimeMicroseconds(timestamp);
+            } catch (IllegalArgumentException e) {
+                LOG.error(
+                        String.format(
+                                "Invalid Timestamp '%s' Provided."
+                                        + "\nShould be a long value indicating microseconds since "
+                                        + "Epoch (1970-01-01 00:00:00.000000+00:00) between %s and %s",
```

Comment:
Just `epoch` is sufficient. Dont have to mention the full value of `1970-01-01 00:00:00.000000+00:00`

Reply:
Fixed.